dpuagent/dpuagent/dpuagent.py

```
@@ -0,0 +1,406 @@
+        self._register_storage_backends()
+
+    def _register_storage_backends(self):
+        raise NotImplementedError
+
+    def get_backend(self, install_path):
+        for prefix, backend_cls in self.backend_registry.items():
+            if install_path.startswith(prefix):
+                return backend_cls()
+
+        raise Exception("No backend driver found for path: %s in vendor driver: %s" %
+                        (install_path, self.__class__.__name__))
```

Ruff (0.15.2)

[warning] Create your own exception

(TRY002)